.github/workflows/create-gui-pr.yml

```diff
@@ -51,7 +51,8 @@ jobs:
         with:
           node-version-file: '.nvmrc'
 
-      - run: yarn install --frozen-lockfile
+      - run: |
+          make clean-install/sync
```

Comment:
This should just be `install/sync` and in the Makefile `install/sync` can then call whatever it likes. The fact that the sync workflow installs 'cleanly' (whatever that means) is controlled in the Makefile itself, which we can have different versions of per branch.

Could we also remove the `clean-install` target, its really confusing and we don't need it. We can just make `install/sync` in turn call `install --frozen-lockfile`.

If this isn't clear or if you have different ideas/plans, could we hook up offline to prevent any further confusion.

Reply:
Oh, well. I’ve changed this now.

For the record, I think it’s a mistake to not distinguish between "install" and "clean-install" explicitly in the command names. It will likely lead to more confusion than just learning the difference between "install" and "clean-install". Now, you will always need to look up what the command actually does and can’t infer it from the command name. In fact, you might assume you can and make the wrong conclusion (e.g. that `make install/sync` runs `npm install` which it doesn’t).

Reply:
Ok cool, I think we should hook up later to make sure we are both on the same page here:

If I'm not mistaken, the additional change here means that `--frozen-lockfile` isn't used when it should be?

https://github.com/kumahq/kuma-gui/blob/8c2b54f8ecfc4c4e078bbff04fcec85ea1be504c/mk/install.mk#L6-L7

https://github.com/kumahq/kuma-gui/blob/8c2b54f8ecfc4c4e078bbff04fcec85ea1be504c/mk/install.mk#L2-L3

---

In my mind `create-gui-sync` should always call `install/sync` and `build/sync` on `master` (which is essentially on all branches). `install/sync` should in turn call `yarn/npm install --frozen-lockfile` (or even `make clean-install`, it doesn't matter really) This way if for some reason a future branch doesn't need a frozen-install or requires something else, we can do that per branch in the Makefile.

If you call the thing that we don't want to change `clean-install` then you are leaking that fact that it does a "clean install" into the name i.e. the thing we are avoiding changing. If we need to change it to not do a clean install, then really we'd need to change the name, and then backport it another 5 times.

All in all its a naming thing, but its similar to when something gets called `blue-button` instead of `primary-button`, and then designers change `blue-button` so its red and we end up with weird confusing naming.

I know its the tiniest of things but, I'd rather get it right now and then forget about it.

Shout me when you are back around.

Reply:
> If I'm not mistaken, the additional change here means that `--frozen-lockfile` isn't used when it should be?

It looks like you’re looking at an outdated version of this branch. This is what the install scripts currently look like:

```sh
.PHONY: install
install:
	npm install

.PHONY: install/sync
install/sync:
	npm clean-install
```

Reply:
Right but the previously backported Makefiles are wrong now aren't they? Because `install/sync` on 2.7 (etc) is like this:

https://github.com/kumahq/kuma-gui/blob/be68aadf480247c57827007cdaa6c8dc0cba6528/mk/install.mk#L1-L7

i.e. without the frozen-lockfile.

The release branches need `frozen-lockfile` don't they?

Reply:
This has all been addressed now so I’m resolving this.